internal/statemachine/classic.go

```diff
@@ -110,17 +104,42 @@ func (stateMachine *StateMachine) parseImageDefinition() error {
 		fmt.Print("WARNING: rootfs.sources-list-deb822 is set to false. The deprecated format will be used to manage sources list. Please if possible adopt the new format.\n")
 	}
 
-	// The official standard for YAML schemas states that they are an extension of
-	// JSON schema draft 4. We therefore validate the decoded YAML against a JSON
-	// schema. The workflow is as follows:
-	// 1. Use the jsonschema library to generate a schema from the struct definition
-	// 2. Load the created schema and parsed yaml into types defined by gojsonschema
-	// 3. Use the gojsonschema library to validate the parsed YAML against the schema
+	err = validateImageDefinition(&imageDefinition)
+	if err != nil {
+		return err
+	}
 
+	classicStateMachine.ImageDef = imageDefinition
+
+	return nil
+}
+
+func readImageDefinition(imageDefPath string) (imagedefinition.ImageDefinition, error) {
+	var imageDefinition imagedefinition.ImageDefinition
+	imageFile, err := os.Open(imageDefPath)
+	if err != nil {
+		return imageDefinition, fmt.Errorf("Error opening image definition file: %s", err.Error())
```

Comment:
In Go the convention (AFAIK) is to ignore other returned values when the error is not nil because they are considered invalid and/or not proplery initialized. But good catch anyway because then we could probably more easily catch we are using an uninitialized object this way. I am more used to pass pointers around in Go but I have not yet "converted" the whole code base yet.